x/emissions/keeper/queryserver/query_server_inferences.go

```diff
@@ -97,10 +100,81 @@ func (qs queryServer) GetLatestNetworkInference(
 		return nil, err
 	}
 
-	return &types.QueryLatestNetworkInferencesAtBlockResponse{
+	return &types.QueryLatestNetworkInferencesResponse{
+		NetworkInferences:         networkInferences,
+		InfererWeights:            synth.ConvertWeightsToArrays(infererWeights),
+		ForecasterWeights:         synth.ConvertWeightsToArrays(forecasterWeights),
+		ForecastImpliedInferences: synth.ConvertForecastImpliedInferencesToArrays(forecastImpliedInferenceByWorker),
+		InferenceBlockHeight:      inferenceBlockHeight,
+		LossBlockHeight:           lossBlockHeight,
+	}, nil
+}
+
+func (qs queryServer) GetLatestAvailableNetworkInference(
+	ctx context.Context,
+	req *types.QueryLatestNetworkInferencesRequest,
+) (
+	*types.QueryLatestNetworkInferencesResponse,
+	error,
+) {
+
+	// Find the latest inference block height
+	_, inferenceBlockHeight, err := qs.k.GetLatestTopicInferences(ctx, req.TopicId)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
You can also add a check here if the blockHeight is equal to zero like in the other queries

Reply:
updated